src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -111,6 +123,64 @@ export default class SourceFileLinter {
 		ts.forEachChild(node, this.#boundVisitNode);
 	}
 
+	analyzeIdentifier(node: ts.Identifier) {
+		const type = this.#checker.getTypeAtLocation(node);
+		if (!type?.symbol || !this.isSymbolOfUi5OrThirdPartyType(type.symbol)) {
+			return;
+		}
+		const deprecationInfo = this.getDeprecationInfo(type.symbol);
+		if (deprecationInfo) {
+			this.#reporter.addMessage({
+				node,
+				severity: LintMessageSeverity.Error,
+				ruleId: RULES["ui5-linter-no-deprecated-api"],
+				message: formatMessage(MESSAGES.SHORT__DEPRECATED_API_ACCESS, node.text),
+				messageDetails: deprecationInfo.messageDetails,
+			});
+		}
+	}
+
+	analyzeImportSpecifier(node: ts.ImportSpecifier) {
+		const type = this.#checker.getTypeAtLocation(node);
+		if (!type?.symbol || !this.isSymbolOfUi5OrThirdPartyType(type.symbol)) {
+			return;
+		}
+		const deprecationInfo = this.getDeprecationInfo(type.symbol);
+		if (deprecationInfo) {
+			this.#reporter.addMessage({
+				node,
+				severity: LintMessageSeverity.Error,
+				ruleId: RULES["ui5-linter-no-deprecated-api"],
+				message: formatMessage(MESSAGES.SHORT__DEPRECATED_API_ACCESS, node.getText()),
+				messageDetails: deprecationInfo.messageDetails,
+			});
+		}
+	}
+
+	analyzeObjectBindingPattern(node: ts.ObjectBindingPattern) {
+		node.elements.forEach((element) => {
+			if (element.kind === ts.SyntaxKind.BindingElement &&
+				element.name.kind === ts.SyntaxKind.Identifier) {
+				this.analyzeIdentifier(element.name);
+			}
+			// TODO: Check whether handling of nested destructuring is required, example:
+			// `const { SomeObject: { SomeOtherObject } } = coreLib;`
```

Comment:
This would be quite a complex task and if we need to handle that, maybe a separate backlog should be created.

It's not only a destructuring, but can become quite complex by messing up destructuring, destructuring with renames, destructuring with array spreading and again destructuring...
As far as I remember we covered this in OpenUI5's JSDoc handler, but I don't think there are currently such usages within UI5

Reply:
Clarified it a bit better in the comment 👍